### PR TITLE
refactor: use vector<u8> for error codes

### PR DESCRIPTION
**Pull request by npty**
# Description

https://axelarnetwork.atlassian.net/browse/AXE-4846

### CODE REVIEW

**Review by Foivos — APPROVED** [review]

File: move/axelar_gateway/sources/discovery.move [hunk 1]
```diff
@@ -14,11 +14,11 @@ use sui::bcs::{Self, BCS};
 use sui::hex;
 use sui::table::{Self, Table};
 
-/// TypeArgument is not a valid string.
-const EInvalidString: u64 = 0;
+#[error]
+const EInvalidString: vector<u8> = b"TypeArgument is not a valid string.";
```
Comment (milapsheth) [inline move/axelar_gateway/sources/discovery.move:18]:
```suggestion
const EInvalidString: vector<u8> = b"typeArgument is not a valid string";
```
remove `.`

File: move/axelar_gateway/sources/gateway.move [hunk 2]
```diff
@@ -49,14 +49,17 @@ const VERSION: u64 = 0;
...
-const EInvalidLength: u64 = 1;
-/// Not latest signers
-const ENotLatestSigners: u64 = 3;
-/// MessageTickets created from newer versions cannot be sent here
-const ENewerMessage: u64 = 4;
+#[error]
+const EMessageNotApproved: vector<u8> = b"Trying to `take_approved_message` for a message that is not approved.";
+
+#[error]
+const EInvalidLength: vector<u8> = b"Invalid message vector length.";
```
Comment (milapsheth) [inline move/axelar_gateway/sources/gateway.move:56]:
```suggestion
const EZeroMessages: vector<u8> = b"no messages found";
```
maybe?

File: move/axelar_gateway/sources/types/proof.move [hunk 3]
```diff
@@ -26,9 +26,14 @@ const SIGNATURE_LENGTH: u64 = 65;
 // Errors
 // ------
 /// Invalid length of the bytes
-const EInvalidLength: u64 = 0;
-const ELowSignaturesWeight: u64 = 1;
-const EMalformedSigners: u64 = 2;
+#[error]
+const EInvalidLength: vector<u8> = b"Invalid signature length: expected 65 bytes";
```
Comment (milapsheth) [inline move/axelar_gateway/sources/types/proof.move:30]:
```suggestion
const EInvalidSignatureLength: vector<u8> = b"Invalid signature length: expected 65 bytes";
```

**Review by milapsheth — APPROVED** [review]

**Comment by milapsheth** [comment]
can you check if it looks fine when an error is returned by the rpc (and perhaps in the explorer if tx is included)?

**Comment by npty** [comment]
> can you check if it looks fine when an error is returned by the rpc (and perhaps in the explorer if tx is included)?

I don't notice any difference. here's my [test tx](https://suiscan.xyz/testnet/tx/FkYJYxDhY2961g8t3u8AkjWmfcK3MchK3GXMepdsRctz), this tx was trying to call `take_approved_message` before the message was approved. Howver, i don't see string error in the block explorer.

**Comment by milapsheth** [comment]
> > can you check if it looks fine when an error is returned by the rpc (and perhaps in the explorer if tx is included)?
> 
> I don't notice any difference. here's my [test tx](https://suiscan.xyz/testnet/tx/FkYJYxDhY2961g8t3u8AkjWmfcK3MchK3GXMepdsRctz), this tx was trying to call `take_approved_message` before the message was approved. Howver, i don't see string error in the block explorer.

Can you ask the mysten team how it could help debug? Is it just more helpful for tests? Or if you run a local sui debug node or something else?

File: move/axelar_gateway/sources/discovery.move [hunk 4]
```diff
@@ -15,10 +15,10 @@ use sui::hex;
 use sui::table::{Self, Table};
 
 #[error]
-const EInvalidString: vector<u8> = b"TypeArgument is not a valid string.";
+const EInvalidString: vector<u8> = b"typeArgument is not a valid string";
```
Comment (milapsheth) [inline move/axelar_gateway/sources/discovery.move:18]:
```suggestion
const EInvalidString: vector<u8> = b"type argument is not a valid string";
```
sorry, camel case still felt weird

File: move/axelar_gateway/sources/gateway.move [hunk 5]
```diff
@@ -51,16 +51,16 @@ const VERSION: u64 = 0;
...
-const EInvalidLength: vector<u8> = b"Invalid message vector length.";
+const EZeroMessages: vector<u8> = b"no mesages found";
 
 #[error]
-const ENotLatestSigners: vector<u8> = b"Not latest signers.";
+const ENotLatestSigners: vector<u8> = b"not latest signers";
 
 #[error]
-const ENewerMessage: vector<u8> = b"MessageTickets created from newer versions cannot be sent here";
+const ENewerMessage: vector<u8> = b"messageTickets created from newer versions cannot be sent here";
```
Comment (milapsheth) [inline move/axelar_gateway/sources/gateway.move:63]:
```suggestion
const ENewerMessage: vector<u8> = b"message ticket created from a newer version cannot be sent";
```

**Review by milapsheth — APPROVED** [review]

**Pull request merged**